scripts/lib/CIME/XML/component.py

```diff
@@ -11,17 +11,22 @@
 
 class Component(EntryID):
 
-    def __init__(self, infile):
+    def __init__(self, infile, comp_class=None):
         """
-        initialize an object
+        initialize a Component obect from the component xml file in infile
+        associate the component class with comp_class if provided.
         """
         files = Files()
         schema = None
-        # not checking schema on external components yet
-        cimeroot = get_cime_root()
-        if  cimeroot in os.path.abspath(infile):
-            schema = files.get_schema("CONFIG_CPL_FILE")
-        EntryID.__init__(self, infile, schema=schema)
+        EntryID.__init__(self, infile)
+        if comp_class is None:
```

Comment:
I think you also need to set self._comp_class in this branch of the conditional. Is it supposed to be set to 'CPL' in this case (I can't tell for sure)?